data/src/main/java/com/linkedin/data/schema/validator/DataSchemaAnnotationValidator.java

```diff
@@ -658,6 +683,21 @@ public void validate(ValidatorContext context)
           }
         }
       }
+      // check if the value belongs to a member in an union and if the member has
+      // validators.
+      if (parentSchema != null && parentSchema.getType() == DataSchema.Type.UNION)
+      {
+        UnionDataSchema unionDataSchema = (UnionDataSchema) parentSchema;
+        Object name = element.getName();
+        if (unionDataSchema.areMembersAliased() && unionDataSchema.contains((String) name))
+        {
+          UnionDataSchema.Member member = unionDataSchema.getMemberByKey((String) name);
+          if (member != null)
```

Comment:
Do you know why this could be null ? (since we checked the schema contains the name?)

Reply:
removed the null check.